### ISSUE TITLE
Connect multiple times with same session 

### ISSUE THREAD

**Issue by ebarriosjr** [hashicorp/boundary#2400]
**Is your feature request related to a problem? Please describe.**
When authenticating a session with a MySQL and trying to add it to an GUI MySQL client like MySQL workbench, most of the people try the connection, but this breaks boundary because it lets you connect only once per session.

**Describe the solution you'd like**
I would like to be able to tell the boundary desktop client to allow multiple connections per session.

**Describe alternatives you've considered**
I thought about telling the developer not to test the connection but if they close and open the GUI connected to the db that will also break. 

**Explain any additional use-cases**
I think reutilization of the same session will be good in many cases. Specially DB connections via anything else than the console.

**Comment by ebarriosjr** [comment]
Oh, I see in the logs the following error:
```
"type":"error",
"data":{
  "error":"no tofu token but not in correct session state",
   "error_fields":{},
   "id":"<redacted>",
   "version":"v0.1",
   "op":"worker.(Worker).handleProxy",
   "info":{"session_id":"redacted"}},
   "datacontentype":"application/cloudevents",
   "time":"2022-08-29T08:05:49.543039096Z"
 }
```
Any ideas what could be causing this? Thanks!

**Comment by ebarriosjr** [comment]
I found an open bug for version 10.1 and thought this was the problem so i upgraded to 10.2 but the error still persists. 
If i run boundary connection from the console instead of from the boundary desktop client i see the following error:
```
error reading handshake result: failed to read protobuf message: failed to get reader: 
received close frame: status = StatusInternalError and reason = "refusing to activate session"
```
And the error on the logs is still the same:
```
no tofu token but in correct session state
```

**Comment by ebarriosjr** [comment]
apparently there is a fix in 10.3. Is there any timeline when 10.3 is coming out?

**Comment by ebarriosjr** [comment]
After locally building version 10.3 i can confirm this fixes the issue.
